notifications/notification_queue.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package notifications
+
+import (
+	"fmt"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/Peripli/service-manager/pkg/types"
+)
+
+var _ NotificationQueue = &notificationQueue{}
+
+// NewNotificationQueue returns new NotificationQueue with specific size
+func NewNotificationQueue(size int) (NotificationQueue, error) {
```

Comment:
Its better (according to the go guidelines) to return concrete type and not interface